configs/run_benchmark.sh

```diff
@@ -1,47 +1,97 @@
 #!/bin/bash
+#
+#
+# Usage:
+# run_benchmark.sh <baremetal|linkerd|istio> <emojivoto|bookinfo> [<rps>] \
+#    [<duration>] [<number of concurrent threads / connections>] \
+#    [<run ID>]
 
-pushgw_base=http://pushgateway.monitoring:9091/metrics/job/wrk2_bench_test
-connections="$((32*3))"
-duration="120"
+pushgw_base=http://pushgateway.monitoring:9091/metrics
+DURATION="600"
 
-app="$1"
+job="$1"
+case "$job" in
+    baremetal)
+        pushgw_base="$pushgw_base/job/bare-metal"
+        ;;
+    linkerd)
+        pushgw_base="$pushgw_base/job/svmesh-linkerd"
+        ;;
+    istio)
+        pushgw_base="$pushgw_base/job/svcmesh-istio"
+        ;;
+    *)
+        echo "Unsupported job '$1'."
+        echo "Supported jobs are 'baremetal', 'linkerd', and 'istio'."
+        exit
+        ;;
+esac
+
+echo "Benchmark job: $job"
+
+
+app="$2"
 case "$app" in
     emojivoto) 
-        pushgw=$pushgw_base/instance/emojivoto
-        rps=25000
+        PUSHGW=$pushgw_base/instance/emojivoto
+        RPS=150000
         app_instance_count=$(kubectl get namespaces | grep emojivoto | wc -l)
         ;;
     bookinfo)
-        pushgw=$pushgw_base/instance/bookinfo
-        rps=3000
+        PUSHGW=$pushgw_base/instance/bookinfo
+        RPS=3000
         app_instance_count=$(kubectl get namespaces | grep bookinfo | wc -l)
         ;;
     *)
         echo "Unsupported app '$1'."
+        echo "Supported apps are 'emojivoto' and 'bookinfo'."
         exit
         ;;
 esac
 
 echo "App: $app"
-echo "RPS: $rps"
+[ -n "$3" ] && RPS="$3"
+
+echo "RPS: $RPS"
+
+[ -n "$4" ] && DURATION="$4"
+echo "Duration: ${DURATION}s"
+
+# a connection / thread can safely handle about 400 concurrent connections
+#  but introduces jitter above that (machine dependent though)
+CONNECTIONS="$(( $RPS / 400 ))"
+[ $CONNECTIONS -lt 1 ] && CONNECTIONS=1
```

Comment:
I didn't quite understand what this means?

Reply:
This is just the thing I didn't get otherwise rest LGTM.

Reply:
A single thread can safely handle up to 400 requests per second. Above that, the RPS start to jitter. So we calculate the default number of threads / connections to be the overall RPS rate divided by 400 - each thread will handle 400 RPS. When the overall RPS rate is below 400, this would result in 0 connections - so we catch that case.